docs/02-Setup.md

```diff
@@ -0,0 +1,252 @@
+   EGO_API=
+   SCOPE=
+
+   # CORS Configuration
+   CORS_ALLOWED_ORIGINS=
+
+   # Vault Configuration (disabled by default)
+   VAULT_ENABLED=false
+   VAULT_URL=http://localhost:8200
+   VAULT_SECRETS_PATH=/kv/lectern
+   VAULT_TOKEN=00000000-0000-0000-0000-000000000000
+   VAULT_ROLE=
+   ```
+
+   <details>
+   <summary><strong>Environment Variables Reference</strong></summary>
+
+   **Express Configuration**
+
+   - `PORT`: Server port (default: 3000)
+   - `OPENAPI_PATH`: Swagger UI path (default: /api-docs)
+
+   **MongoDB Configuration**
+
+   - `MONGO_HOST`: Database hostname (default: localhost)
+   - `MONGO_PORT`: Database port (default: 27017)
+   - `MONGO_DB`: Database name (default: lectern)
+   - `MONGO_USER`: Database username (optional)
+   - `MONGO_PASS`: Database password (optional)
+
+   **Authentication (Optional)**
+
+   - `AUTH_ENABLED`: Enable JWT-based authorization (default: false)
+   - `EGO_API`: EGO API URL for JWT validation
+   - `SCOPE`: Required policy name in JWT scope
+   - `CORS_ALLOWED_ORIGINS`: Comma-separated list of allowed origins
+
+   **Vault Integration (Optional)**
+
+   - `VAULT_ENABLED`: Enable HashiCorp Vault integration (default: false)
+   - `VAULT_URL`: Vault server URL
+   - `VAULT_SECRETS_PATH`: Path to secrets in Vault
+   - `VAULT_TOKEN`: Vault access token
+   - `VAULT_ROLE`: Vault role for authentication
+
+   </details>
+
+4. **Build the Application**
+
+   ```bash
+   # From workspace root
+   pnpm nx build @overture-stack/lectern-server
+
+   # Or from apps/server directory
+   pnpm build
+   ```
+
+5. **Start the Development Server**
+
+   ```bash
+   # Production mode
+   pnpm nx start @overture-stack/lectern-server
+
+   # Development mode with hot reloading
+   pnpm nx debug server
+   ```
+
+## Verification & Testing
+
+### API Health Check
+
+Verify that Lectern is running correctly:
+
+```bash
+# Health endpoint
+curl http://localhost:3000/health
+
+# Expected response: 200 OK
+```
+
+### API Documentation
+
+Access the interactive API documentation at:
+
+- **Swagger UI**: `http://localhost:3000/api-docs`
+
+### Dictionary Management Testing
+
+1. Navigate to the Swagger UI
+2. Test creating a new data dictionary using the REST API
+3. Verify dictionary creation, retrieval, and management operations
+
+**Troubleshooting:**
+
+- Ensure MongoDB is running and accessible
+- Check server logs for validation errors
+- Verify API endpoints are responding correctly
+
+:::info Need Help?
+If you encounter any issues or have questions about our API, please don't hesitate to reach out through our relevant [**community support channels**](https://docs.overture.bio/community/support).
+:::
+
+## Advanced Configuration
+
+### Enabling Authorization
+
+For production environments, enable JWT-based authorization:
+
+1. Set `AUTH_ENABLED=true` in your `.env` file
+2. Configure `EGO_API` to point to your Ego authorization service
+3. Set the appropriate `SCOPE` for your permissions
```

Comment:
This is unclear, maybe we need to link to a reference that explains scopes from Ego/Keycloak. I wouldn't know what value to set in this.